lib/inventory/metadata/metadata.go

```diff
@@ -210,14 +199,35 @@ func (c *fetchConfig) fetchContainerRuntime() string {
 
 // fetchContainerOrchestrator returns kubernetes-${GIT_VERSION} if the instance is
 // running on kubernetes.
+// This function performs the equivalent of the following:
+// curl -k https://$KUBERNETES_SERVICE_HOST:$KUBERNETES_SERVICE_PORT/version | jq .gitVersion
 func (c *fetchConfig) fetchContainerOrchestrator() string {
-	if c.kubeClient == nil {
+	url := fmt.Sprintf("https://%s:%s/version", c.getenv("KUBERNETES_SERVICE_HOST"), c.getenv("KUBERNETES_SERVICE_PORT"))
+	req, err := http.NewRequestWithContext(c.context, http.MethodGet, url, nil)
+	if err != nil {
+		log.Debugf("Failed to create kubernetes http GET request '%s': %s", url, err)
+		return ""
```

Comment:
Should we return the error to the caller? They can still choose to ignore errors and proceed, but it makes this method less ambiguous.

Reply:
All `fetch*` methods in this package follow this approach and I would prefer not changing them all now.